### PR TITLE
feat(api): Update previous pool points in user points

### PR DESCRIPTION
**Pull request by rohanjadvani**
## Summary

Backfill pool 1, 2, and 3 points when refreshing users

## Testing Plan

Unit test

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: src/user-points/user-points.jobs.controller.ts [hunk 1]
```diff
@@ -68,10 +74,69 @@ export class UserPointsJobsController {
     }
 
     const options = await this.eventsService.getUpsertPointsOptions(user);
-    await this.userPointsService.upsert(options);
+    const userPoints = await this.userPointsService.upsert(options);
+
+    if (
```
Comment (jowparks) [inline src/user-points/user-points.jobs.controller.ts:79]: Won't these default to 0 based on last PR?

Comment (rohanjadvani) [reply src/user-points/user-points.jobs.controller.ts:79]: They are nullable: https://github.com/iron-fish/ironfish-api/pull/1275/files#diff-5b443964f4f3a611682db8f7e02177b0a8c632b2039e2bd5e4dd7347815c565cR265-R267

File: src/user-points/user-points.jobs.controller.ts [hunk 2]
```diff
@@ -68,10 +74,69 @@ export class UserPointsJobsController {
...
+      userPoints.pool1_points === null ||
+      userPoints.pool2_points === null ||
+      userPoints.pool3_points === null
+    ) {
+      const refreshPreviousPoolOptions: RefreshPreviousPoolOptions = {
```
Comment (jowparks) [inline src/user-points/user-points.jobs.controller.ts:84]: Shouldn't we be getting points for pull requests from all phases?

Comment (rohanjadvani) [reply src/user-points/user-points.jobs.controller.ts:84]: We have points for pull requests from previous phases in the current db

**Review by jowparks — APPROVED** [review]

**Pull request merged**